packages/twenty-front/src/modules/settings/accounts/utils/computeSyncStatus.ts

```diff
@@ -0,0 +1,42 @@
+import { CalendarChannelSyncStatus } from '@/accounts/types/CalendarChannel';
+import { MessageChannelSyncStatus } from '@/accounts/types/MessageChannel';
+import { SyncStatus } from '@/settings/accounts/enums/syncStatus.enum';
+
+export const computeSyncStatus = (
+  messageChannelSyncStatus: MessageChannelSyncStatus,
+  calendarChannelSyncStatus: CalendarChannelSyncStatus,
+) => {
+  if (
+    messageChannelSyncStatus === MessageChannelSyncStatus.FAILED_UNKNOWN ||
+    messageChannelSyncStatus ===
+      MessageChannelSyncStatus.FAILED_INSUFFICIENT_PERMISSIONS ||
+    calendarChannelSyncStatus === CalendarChannelSyncStatus.FAILED_UNKNOWN ||
+    calendarChannelSyncStatus ===
+      CalendarChannelSyncStatus.FAILED_INSUFFICIENT_PERMISSIONS
+  ) {
+    return SyncStatus.FAILED;
+  }
+
+  if (
+    messageChannelSyncStatus === MessageChannelSyncStatus.ONGOING ||
+    calendarChannelSyncStatus === CalendarChannelSyncStatus.ONGOING
+  ) {
+    return SyncStatus.IMPORTING;
+  }
+
+  if (
+    messageChannelSyncStatus === MessageChannelSyncStatus.NOT_SYNCED ||
```

Comment:
I think this is wrong, in case of both NOT_SYNCED it should be NOT_SYNC
In case of one out of 2 NOT_SYNCED we should leverage the other one (which is actually the case because all other conditions are or, except the last one ACTIVE)